pkg/platform/runtime2/artifact/download.go

```diff
@@ -15,8 +14,12 @@ type ArtifactDownload struct {
 	Checksum    string
 }
 
+type urlSigner interface {
+	SignS3URL(uri *url.URL) (*url.URL, error)
+}
+
 // NewDownloadsFromBuild extracts downloadable artifact information from the build status response
-func NewDownloadsFromBuild(buildStatus *headchef_models.BuildStatusResponse) ([]ArtifactDownload, error) {
+func NewDownloadsFromBuild(model urlSigner, buildStatus *headchef_models.BuildStatusResponse) ([]ArtifactDownload, error) {
```

Comment:
Nit: I'd prefer not doing this again unless we really have to. Just have `NewDownloadsFromBuild` call a new function and then do the signing with the results of that function. Then you'd just have to unit test the new function that has nothing to do with signing.

Reply:
I like doing the signing outside of this function yes.

Reply:
Turns out, we *had* to sign later, because if we use the build log streamer, the URL were never signed as it was!  Good that we caught it. :)